ui/src/app/dashboard/views/DashboardConfigurationForm/DashboardConfigurationSubnetForm/DashboardConfigurationSubnetForm.tsx

```diff
@@ -0,0 +1,115 @@
+import FormikForm from "app/base/components/FormikForm";
+import LegacyLink from "app/base/components/LegacyLink";
+import baseURLs from "app/base/urls";
+import { actions as fabricActions } from "app/store/fabric";
+import fabricSelectors from "app/store/fabric/selectors";
+import { actions as subnetActions } from "app/store/subnet";
+import subnetSelectors from "app/store/subnet/selectors";
+import type { Subnet } from "app/store/subnet/types";
+
+type SubnetDiscoveryValues = {
+  [x: number]: Subnet["active_discovery"];
+};
+
+const DashboardConfigurationSubnetForm = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const subnets = useSelector(subnetSelectors.all);
+  const fabrics = useSelector(fabricSelectors.all);
+  const subnetsLoaded = useSelector(subnetSelectors.loaded);
+  const fabricsLoaded = useSelector(subnetSelectors.loaded);
+  const saved = useSelector(subnetSelectors.saved);
+  const saving = useSelector(subnetSelectors.saving);
+
+  useEffect(() => {
+    dispatch(subnetActions.fetch());
+    dispatch(fabricActions.fetch());
+  }, [dispatch]);
+
+  const loaded = subnetsLoaded && fabricsLoaded;
+  let content: JSX.Element = <Spinner text="Loading..." />;
+
+  if (loaded) {
+    const initialValues: SubnetDiscoveryValues = {};
+    subnets.forEach((subnet) => {
+      initialValues[subnet.id] = subnet.active_discovery;
+    });
+
+    content = (
+      <FormikForm<SubnetDiscoveryValues>
+        buttonsAlign="left"
+        buttonsBordered={false}
+        initialValues={initialValues}
+        onSubmit={(values, { resetForm }) => {
+          subnets.forEach((subnet) => {
+            if (subnet.active_discovery !== values[subnet.id]) {
+              dispatch(
+                subnetActions.update({
+                  active_discovery: values[subnet.id],
+                  id: subnet.id,
+                })
+              );
+            }
+          });
+          resetForm({ values });
+        }}
+        resetOnSave
+        saving={saving}
+        saved={saved}
+      >
+        {subnets.map((subnet) => {
+          const targetFabric = fabrics.find((fabric) =>
+            fabric.vlan_ids.includes(subnet.vlan)
+          );
+          return (
+            <FormikField
+              key={`subnet-${subnet.id}`}
+              label={
+                <>
+                  <LegacyLink
+                    data-test="subnet-link"
+                    route={baseURLs.subnet({ id: subnet.id })}
+                  >
+                    {subnet.cidr}
+                  </LegacyLink>
+                  {targetFabric && (
+                    <>
+                      <span> on </span>
+                      <LegacyLink
+                        data-test="fabric-link"
+                        route={baseURLs.fabric({ id: targetFabric.id })}
+                      >
+                        {targetFabric.name}
+                      </LegacyLink>
+                    </>
+                  )}
```

Comment:
@petesfrench We use this `LegacyLink` component for all legacy links. Just means we don't have to hardcode in the whole routes. Also I've made the fabric link bit conditional, in the off chance the subnet's fabric can't be found from the list of fabrics.